### ISSUE TITLE
Documentation improvements

### ISSUE THREAD

**Issue by nylen** [seb86/WordPress-REST-API-jQuery#3]
For example:

- When I make a call to an endpoint, how do I get the results?
- What format are these results in?

**Comment by seb86** [comment]
Documentation will be provided once the script is complete. Code is not finalised at this time.

**Comment by nylen** [comment]
This sounds like a good idea to avoid duplicating work, but in practice this usually leads to documentation never being written.  Also, code is never "complete", and the process of writing documentation often informs design improvements.  (Similarly with unit tests, but that is another topic.)

**Comment by seb86** [comment]
After the next update I will start writing docmentation. Most likely only be for _GET_ requests until [authentication](https://github.com/seb86/WordPress-REST-API-jQuery/issues/4) is figured out.

**Comment by seb86** [comment]
I have started to write the documentation using [Slate](https://github.com/tripit/slate). Nothing much at the moment. Just getting the structure ready.

https://seb86.github.io/WordPress-REST-API-jQuery-Documentation/

**Comment by nylen** [comment]
I think this project is small enough that the documentation could just live in the readme.  In any case you want to avoid duplicating information. If you're using slate I guess you could replace the readme with a few sentences + a link to the documentation site?  Though my opinion is still that it's not worth making users go to another site for documentation.

It's still not clear how to receive the results of an API call. Does `restjQuery` accept a callback function? Return a promise? This should be documented and indicated in the examples.

**Comment by seb86** [comment]
I will provide clear examples soon.

**Comment by nylen** [comment]
> When I make a call to an endpoint, how do I get the results?

I found the answer to this in the code rather than the documentation:  this library uses `async: false`, so when a request is made, the data is returned immediately.

You should change this as soon as possible to use promises and/or callbacks.  Here's why:

- http://stackoverflow.com/a/34813588/106302
- https://xhr.spec.whatwg.org/#synchronous-flag

> Synchronous `XMLHttpRequest` outside of workers is in the process of being removed from the web platform as it has detrimental effects to the end user’s experience. (This is a long process that takes many years.) Developers **must not** pass `false` for the `async` argument when current global object is a `Window` object. User agents are strongly encouraged to warn about such usage in developer tools and may experiment with throwing an `InvalidAccessError` exception when it occurs.

**Comment by seb86** [comment]
Yes I have been reading these types of topics to improve the library. I have little time at the moment to apply these improvements but will try to get it done soon. I am in the middle of writing the documentation currently.
